Kitodo/src/main/java/org/kitodo/production/services/data/ProcessService.java

```diff
@@ -581,454 +489,162 @@ public void refresh(Process process) {
-     * moment. This will be aborted after a timeout so that the user gets an
-     * answer (which may be incomplete) in finite time.
-     *
-     * @param searchInput
-     *            user input
-     * @param rulesetId
-     *            the id of the allowed ruleset
-     * @param allowedStructuralElementTypes
-     *            allowed topmost logical structural elements
-     * @return found processes
-     * @throws DataException
-     *             if the search engine fails
-     */
-    public List<ProcessDTO> findLinkableChildProcesses(String searchInput, int rulesetId,
+    @Override
+    public List<Process> findLinkableChildProcesses(String searchInput, int rulesetId,
             Collection<String> allowedStructuralElementTypes) throws DataException {
 
-        BoolQueryBuilder query = new BoolQueryBuilder()
-                .must(new BoolQueryBuilder()
-                        .should(new MatchQueryBuilder(ProcessTypeField.ID.getKey(), searchInput).lenient(true))
-                        .should(new WildcardQueryBuilder(ProcessTypeField.TITLE.getKey(), "*" + searchInput + "*")))
-                .must(new MatchQueryBuilder(ProcessTypeField.RULESET.getKey(), rulesetId));
-        List<ProcessDTO> linkableProcesses = new LinkedList<>();
-
-        List<ProcessDTO> processDTOS = findByQuery(query, false);
-        for (ProcessDTO process : processDTOS) {
-            if (allowedStructuralElementTypes.contains(getBaseType(process.getId()))) {
-                linkableProcesses.add(process);
-            }
-        }
-        return linkableProcesses;
+        BeanQuery query = new BeanQuery(Process.class);
+        query.restrictToClient(ServiceManager.getUserService().getSessionClientId());
+        query.addNullRestriction("parent_id");
+        query.addIntegerRestriction("ruleset_id", rulesetId);
+        query.forIdOrInTitle(searchInput);
+        return getByQuery(query.formQueryForAll(), query.getQueryParameters());
     }
 
-    /**
-     * Searches for linkable processes based on user input. A process can be
-     * linked if it has the same rule set, belongs to the same client, and the
-     * topmost element of the logical outline below the selected parent element
-     * is an allowed child. For the latter, the data file must be read at the
-     * moment. This will be aborted after a timeout so that the user gets an
-     * answer (which may be incomplete) in finite time.
-     *
-     * @param searchInput
-     *            user input
-     * @param projectId
-     *            the id of the allowed project
-     * @param rulesetId
-     *            the id of the allowed ruleset
-     * @return found processes
-     * @throws DataException
-     *             if the search engine fails
-     */
-    public List<ProcessDTO> findLinkableParentProcesses(String searchInput, int projectId, int rulesetId)
+    @Override
+    public List<Process> findLinkableParentProcesses(String searchInput, int projectId, int rulesetId)
```

Comment:
This behaviour will be changed in release 3.7.0: https://github.com/kitodo/kitodo-production/pull/5869 
If the logic gets changed or re-implemented here, the concept of the linked PR should better guide the implementation.

Reply:
Question. Do I understand correctly that this is just a hint, it is not a request to change anything in this pull request?

Reply:
When @solth merges the changes of https://github.com/kitodo/kitodo-production/pull/5869 into master, which is happening soon, it would probably be the best to rebase your changes based on the changes there. (After merging of #5869 it will be possible to link to processes in another project so the search for parents should include processes in other projects). If you prefer continue working on the concept in current master (search happens only in the same project) the merge with the new changes has to happen later. I guess it depends on what you prefer.

Reply:
As long as the changes have not been merged into the branch I developed them for, I cannot merge master into the current branch. I have to wait and see, I think. The merge has to happen at some point, that's clear. So, thanks for the explanation.